File: PO_Projekt3/guarana.py
from roslina import Roslina

class Guarana(Roslina):
    def __init__(self, swiat, polozenie):
        super().__init__(swiat, polozenie, 0, 0, 0.10, 1)

    def kolizja(self, ofiara, atak):
        if atak:
            return

        ofiara.setSila(ofiara.getSila() + 3)

        if self._zyje:
            self.zakonczZycie()

    def tegoSamegoTypu(self, inny):
        return isinstance(inny, Guarana)

    def stworzDziecko(self, cel):
        return Guarana(self._swiat, cel)

    def rysowanie(self):
        return (255, 102, 217)

File: projekty/PO_Projekt2/ProgOb_Projekt2/src/main/java/pl/Dybowski/Adrian/Projekt2/WirtualnySwiat/Rosliny/Guarana.java
package pl.Dybowski.Adrian.Projekt2.WirtualnySwiat.Rosliny;

import pl.Dybowski.Adrian.Projekt2.WirtualnySwiat.Organizm;
import pl.Dybowski.Adrian.Projekt2.WirtualnySwiat.Stale;
import pl.Dybowski.Adrian.Projekt2.WirtualnySwiat.Swiat;
import pl.Dybowski.Adrian.Projekt2.WirtualnySwiat.Wspolrzedne2D;

public class Guarana extends Roslina {

	public Guarana(Swiat swiat, Wspolrzedne2D polozenie) {
		super(swiat, polozenie, Stale.sila.get(Guarana.class), Stale.inicjatywa.get(Guarana.class),
				Stale.szansaRozprzestrzeniania.get(Guarana.class), Stale.probyRozprzestrzeniania.get(Guarana.class));
	}
	
	@Override
	public void kolizja(Organizm ofiara, boolean atak) {
		if (atak) return;
		
		ofiara.zmienSila(3);
		
		if (zyje) zakonczZycie();
	}

	@Override
	public boolean tegoSamegoTypu(Organizm inny) {
		return inny instanceof Guarana;
	}

	@Override
	public Organizm stworzDziecko(Wspolrzedne2D cel) {
		return new Guarana(swiat, cel);
	}

}
